fixtures/cloudflare-dev-module-resolution/index.test.ts

```diff
@@ -0,0 +1,50 @@
+import { createServer } from 'vite';
+import { beforeAll, describe, expect, test } from 'vitest';
+
+import config from './vite.config';
+
+describe('Cloudflare dev module resolution', () => {
+  let jsonOutput: Record<string, unknown> = {};
+
+  beforeAll(async () => {
+    jsonOutput = (await fetchJsonFromViteDevServer()) ?? {};
+  });
+
+  test('can successfully import "React"', () => {
+    expect(jsonOutput['typeof React']).toEqual('object');
+    expect(jsonOutput['typeof React.cloneElement']).toEqual('function');
+    expect(jsonOutput['reactVersionsMatch']).toEqual(true);
+  });
+
+  test('can successfully import utilities from "@remix-run/cloudflare"', () => {
+    expect(jsonOutput['type of remix cloudflare json({})']).toEqual('object');
+    expect(jsonOutput['remixRunCloudflareCookieName']).toEqual(
+      'my-remix-run-cloudflare-cookie',
+    );
+  });
+});
+
+async function fetchJsonFromViteDevServer(): Promise<null | Record<
+  string,
+  unknown
+>> {
+  const viteServer = await createServer({
+    ...config,
+  });
+
+  await viteServer.listen();
+
+  const addressInfo = viteServer.httpServer?.address();
+
+  if (!addressInfo) {
+    return null;
+  }
```

Comment:
Is it better to throw an error here?

Reply:
I have no preference, if things break you'll see the tests failing either way 🙂

if you prefer I can just throw here

Reply:
Yes, it will make it easier to know why it failed:

```suggestion
  if (!addressInfo) {
    throw new Error("addressInfo not found");
  }
```